apollo-client/src/main/java/com/ctrip/framework/apollo/spring/config/PropertySourcesProcessor.java

```diff
@@ -55,22 +56,35 @@
  */
 public class PropertySourcesProcessor implements BeanFactoryPostProcessor, EnvironmentAware,
     ApplicationEventPublisherAware, PriorityOrdered {
-  private static final Multimap<Integer, String> NAMESPACE_NAMES = LinkedHashMultimap.create();
+  private static final Multimap<Integer, Multimap<String,String>> APP_NAMESPACE_NAMES = LinkedHashMultimap.create();
   private static final Set<BeanFactory> AUTO_UPDATE_INITIALIZED_BEAN_FACTORIES = Sets.newConcurrentHashSet();
 
   private final ConfigPropertySourceFactory configPropertySourceFactory = SpringInjector
       .getInstance(ConfigPropertySourceFactory.class);
-  private ConfigUtil configUtil;
+  private ConfigUtil configUtil = ApolloInjector.getInstance(ConfigUtil.class);
   private ConfigurableEnvironment environment;
   private ApplicationEventPublisher applicationEventPublisher;
 
+  public PropertySourcesProcessor(){
+    if(configUtil == null){
+      configUtil = ApolloInjector.getInstance(ConfigUtil.class);
+    }
+  }
+
   public static boolean addNamespaces(Collection<String> namespaces, int order) {
-    return NAMESPACE_NAMES.putAll(order, namespaces);
+    ConfigUtil configUtil = ApolloInjector.getInstance(ConfigUtil.class);
```

Comment:
The file contains numerous instances of `configUtil` initialization, which isn't ideal.